### DISCUSSION TITLE
say instructions, which one do we really need?

### DISCUSSION THREAD

**Discussion by Peter-Jacob** [adesutherland/CREXX#288] [Q&A]
while implementing SAYX I discovered that we have several say statements integer/float/ character ... During my testing I realized we really need just 2 SAY_STRING and SAY_REG. It seems the REXX compiler translates them only in the two. 
Am I correct? If so shouldn't we remove the other ones?

**Comment by rvjansen** [comment]
In my opinion, these overloaded instructions can be useful for the rxas assembler programmer, even if the compiler does not generate them, they are not in the way, so for me they can stay. If at some point someone misses them, it is quite a job to put them back.

**Comment by rvjansen** [comment]
Looking at the instruction reference (that I am putting together) I see what you mean now. We should either offer SAY and SAYX in all overloaded versions, or remove them on SAY. I currently lean to removal of the superfluous ones on SAY.

What SAY you?

**Comment by rvjansen** [comment]
and this apart from the fact we don't have a say `decimal`:
```
0x00d9 say        {REG}                Say op1
0x00dc say        {INT}                Say op1
0x00dd say        {FLOAT}              Say op1
0x00df say        {CHAR}               Say op1
0x00de say        {STRING}             Say op1
0x00da sayx       {REG}                Say op1 without line feed
0x00db sayx       {STRING}             Say op1 (as string) without line feed
```

**Comment by rvjansen** [comment]
and I gather that it would be possible (when a register represents an object of a class) to call a toString() function when using SAY/SAYX?

**Comment by adesutherland** [comment]
The say float etc. are for the benefit of assembler programmes. If they can
be bothered to use ftos ...
